discord/ext/commands/cog.py

```diff
@@ -585,6 +585,18 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
         """
         return True
 
+    @_cog_special_method
+    def interaction_check(self, interaction: discord.Interaction[BotT], /) -> bool:
```

Comment:
This PR is failing because `BotT` doesn't quite match `ClientT` so you'll have to use the latter.